src/map/skill.cpp

```diff
@@ -2624,32 +2618,133 @@ int skill_break_equip(struct block_list *src, struct block_list *bl, unsigned sh
 	return where; //Return list of pieces broken.
 }
 
-int skill_strip_equip(struct block_list *src,struct block_list *bl, unsigned short where, int rate, int lv, int time)
+/**
+ * Strip equipment from a target
+ * @param src: Source of call
+ * @param target: Target to strip
+ * @param skill_id: Skill used
+ * @param skill_lv: Skill level used
+ * @return True on successful strip or false otherwise
+ */
+bool skill_strip_equip(struct block_list *src, struct block_list *target, uint16 skill_id, uint16 skill_lv)
 {
-	struct status_change *sc;
+	nullpo_retr(false, src);
+	nullpo_retr(false, target);
+
+	struct status_change *tsc = status_get_sc(target);
+
+	if (!tsc || tsc->option&OPTION_MADOGEAR) // Mado Gear cannot be divested [Ind]
+		return false;
+
 	const int pos[5]             = {EQP_WEAPON, EQP_SHIELD, EQP_ARMOR, EQP_HELM, EQP_ACC};
 	const enum sc_type sc_atk[5] = {SC_STRIPWEAPON, SC_STRIPSHIELD, SC_STRIPARMOR, SC_STRIPHELM, SC__STRIPACCESSORY};
 	const enum sc_type sc_def[5] = {SC_CP_WEAPON, SC_CP_SHIELD, SC_CP_ARMOR, SC_CP_HELM, SC_NONE};
-	int i;
+	struct status_data *sstatus = status_get_status_data(src), *tstatus = status_get_status_data(target);
+	int rate, time, location;
+
+	switch (skill_id) { // Rate
+		case RG_STRIPWEAPON:
+		case RG_STRIPARMOR:
+		case RG_STRIPSHIELD:
+		case RG_STRIPHELM:
+		case GC_WEAPONCRUSH:
+			rate = 50 * (skill_lv + 1) + 2 * (sstatus->dex - tstatus->dex);
+			break;
+		case ST_FULLSTRIP: {
+			int min_rate = 50 + 20 * skill_lv;
+
+			rate = min_rate + 2 * (sstatus->dex - tstatus->dex);
+			rate = min(min_rate, rate);
```

Comment:
wait shouldnt it be max?

Reply:
You're right! I was in the "minimum" mindset. 🙈